crates/sgx_validation/src/config.rs

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use serde::de::Error;
+use serde::{Deserialize, Deserializer, Serialize};
+use sgx::parameters::Features;
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    /// Values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary which are disallowed.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `features`.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_features")]
+    pub features: Option<u64>,
+
+    /// Minimum value allowed for `cpusvn`.
+    pub cpu_svn: Option<Vec<u8>>,
+
+    /// Minimum value for `isv_svn`.
+    pub enclave_security_version: Option<u16>,
```

Comment:
Why not just `min_isv_svn`?

Reply:
1. Because it's compared against `rpt.enclave_product_id()` from `sgx::ReportBody` in the `sgx` crate.
2. I wanted something more user friendly, more descriptive for the config file so whomever uses this has a better idea of that's going on. By comparison, `isv_svn` doesn't describe what it represents.

Reply:
Either the field name or the comment has to be changed.
By looking at this field and the comment, it's absolutely not clear how `isv_svn` and a security version of the enclave (what is that at all?) are related